src/lib/state/network/NetworkSimulationEvent.ts

```diff
@@ -0,0 +1,29 @@
+import Signal, { Connection } from "../../utilities/simple-signal";
+import { NetworkEvent } from "./NetworkEvent";
+
+const SimulatedPlayer = { Name: "NetworkSimulation" } as Player;
+
+export class NetworkSimulationEvent extends NetworkEvent {
+	private serverSignal = new Signal<[Player, ...Array<unknown>]>();
```

Comment:
I *might* disagree on using another implementation of a Signal module other than the native one at Tina (an `EventListener`, if any doubts on how to implement it, let me know and I can myself do it, or you can look thru Networking which uses the EventListeners internally to bind the callbacks, etc.)

Reply:
I could do this but I would need to change EventEmitter to an instantiable class since it's also used in the ObservableValue class. I wouldn't want to extend it since the ObservableValue can only be subscribed to and not emitted outside of the class.